docs/community/contribute/checks/overview.md

```diff
@@ -0,0 +1,122 @@
+
+Before writing a new check, you need to verify if the provider your check targets is supported by Trivy. If it's not, you'll need to add support for it. Additionally, if the provider that you want to target exists, you need to check whether the service your policy will target is supported.
+
+### Add Support for a New Cloud Provider
+
+[Please reference the documentation on adding Support for a New Cloud Provider](./provider-support.md).
+
+### Add Support for a New Service in an existing Provider
+
+[Please reference the documentation on adding Support for a New Service](./service-support.md).
+
+This guide also showcases how to add new properties for an existing Service.
+
+## Check Metadata
+
+The metadata is the top section that starts with `# METADATA`, and has to be placed on top of the check. You can copy and paste from another check as a starting point. This format is effectively _yaml_ within a Rego comment, and is [defined as part of Rego itself](https://www.openpolicyagent.org/docs/latest/policy-language/#metadata).
+
+The metadata consists of the following fields:
+
+- `title` is a title for the rule. The title should clearly and succinctly state the problem which is being detected.
+- `description` is a description of the problem which is being detected. The description should be a little more verbose than the title, and should describe what the rule is trying to achieve. Imagine it completing a sentence starting with `You should...`.
+- `scope` is used to define the scope of the policy. In this case, we are defining a check that applies to the entire package. At the moment, Trivy only supports using package scope for metadata, so this should always be set to `package`.
+- `schemas` tells Rego that it should use the `AWS` schema to validate the use of the input data in the policy. We currently support [the following shemas](https://github.com/aquasecurity/trivy/tree/main/pkg/iac/rego/schemas). Please choose a schema that is applicable to your check. Using a schema can help you validate your policy faster for syntax issues.
+- `custom` is used to define custom fields that can be used by Trivy to provide additional context to the policy and any related detections. This can contain the following:
+    - `avd_id` can be used to link the check to the AVD entry. In the example check above, the `avd_id` `AVD-AWS-0176` is the ID of the check in the [AWS Vulnerability Database](https://avd.aquasec.com/). If there is no AVD_ID available, you need to generate an ID to use for this field using `make id`.
+    - `provider` is the name of the [provider](https://github.com/aquasecurity/defsec/tree/master/pkg/providers) the check targets. This should be the same as the provider name in the `pkg/providers` directory, e.g. `aws`.
+    - `service` is the name of the service by the provider that the check targets. This should be the same as the service name in the `pkg/providers` directory, e.g. `rds`.
```

Comment:
I think this is relevant only for cloud checks? also I remember saying sometime that this is overlapping with selector, did we address that?  (@simar7 )

Reply:
Yes. This is optional and only needed if you want to optimize your scanning. See my comment [here](https://github.com/aquasecurity/trivy/pull/6234/files/c549765e504d59cb2ba80cb55655979a83cf8e4c#r1511596970) for what's optional and required.